src/pkg/boltz/boltz.ts

```diff
@@ -0,0 +1,120 @@
+// docs: https://docs.boltz.exchange/en/latest/api/
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import axios, { AxiosError } from 'axios';
+import { extractErrorMessage } from '../../extension/utility/error';
+
+export type NetworkString = 'liquid' | 'testnet' | 'regtest';
+
+interface CreateSwapCommonRequest {
+  type: 'submarine' | 'reversesubmarine';
+  pairId: 'L-BTC/BTC';
+  orderSide: 'buy' | 'sell';
+}
+
+export interface CreateSwapCommonResponse {
+  id: string;
+  timeoutBlockHeight: number;
+}
+
+export type SubmarineSwapRequest = {
+  invoice: string;
+  refundPublicKey: string;
+};
+
+export type SubmarineSwapResponse = {
+  acceptZeroConf: boolean;
+  address: string;
+  bip21: string;
+  blindingKey: string; // private
+  expectedAmount: number;
+  redeemScript: string;
+};
+
+export type ReverseSubmarineSwapRequest = {
+  preimageHash: string;
+  invoiceAmount: number;
+  claimPublicKey: string;
+};
+
+export type ReverseSubmarineSwapResponse = {
+  blindingKey: string; // private
+  invoice: string;
+  lockupAddress: string;
+  onchainAmount: number;
+  redeemScript: string;
+};
+
+export const boltzUrl: Record<NetworkString, string> = {
+  regtest: 'http://localhost:9090',
+  testnet: 'https://testnet.boltz.exchange/api',
+  liquid: 'https://boltz.exchange/api',
+};
+
+//
+export class Boltz {
```

Comment:
This cannot be "merged" with `BoltzService` class ? It seems only useful there

Reply:
Boltz is only used by BoltzService indeed. But I found it cleaner to separate the Boltz API and the Boltz application API. You sure you don't agree?